packages/did-auth-siop-op-authenticator/README.md

```diff
@@ -0,0 +1,214 @@
+<!--suppress HtmlDeprecatedAttribute -->
+<h1 align="center">
+  <br>
+  <a href="https://www.sphereon.com"><img src="https://sphereon.com/content/themes/sphereon/assets/img/logo.svg" alt="Sphereon" width="400"></a>
+  <br>DID Auth SIOP OP Authenticator (Typescript) 
+  <br>
+</h1>
+
+---
+
+**Warning: This package still is in every early development. Breaking changes without notice will happen at this point!**
+
+---
+
+A Veramo operating party authentication plugin using the [Self Issued OpenID Provider v2 (SIOP)](https://github.com/Sphereon-Opensource/did-auth-siop) authentication library for having clients / people conforming to
+the [Self Issued OpenID Provider v2 (SIOPv2)](https://openid.net/specs/openid-connect-self-issued-v2-1_0.html)
+and  [OpenID Connect for Verifiable Presentations (OIDC4VP)](https://openid.net/specs/openid-connect-4-verifiable-presentations-1_0.html)
+as specified in the OpenID Connect working group.
+
+## Self Issued OpenID Provider v2 (SIOP)
+
+For more information about [Self Issued OpenID Provider v2 (SIOP)](https://github.com/Sphereon-Opensource/did-auth-siop#introduction), see the documentation in the readme.
+
+## Requirements
+For this plugin a DID resolver is also required. A DID resolver can be added to the agent as plugin as seen in the example below.
+
+## Available functions
+
+* getDidSiopSession
+* addDidSiopSession
+* removeDidSiopSession
+* registerCustomApprovalForDidSiop
+* removeCustomApprovalForDidSiop
+* authenticateWithDidSiop
+* getDidSiopAuthenticationRequestFromRP
+* getDidSiopAuthenticationRequestDetails
+* verifyDidSiopAuthenticationRequestURI
+* sendDidSiopAuthenticationResponse
+
+The following functions can also be used on the session object without the need of a session id first.
+
+* authenticateWithDidSiop
+* getDidSiopAuthenticationRequestFromRP
+* getDidSiopAuthenticationRequestDetails
+* verifyDidSiopAuthenticationRequestURI
+* sendDidSiopAuthenticationResponse
+
+## Usage
+
+### Adding the plugin to an agent:
+
+```js
+import { IDidAuthSiopOpAuthenticator } from '@sphereon/ssi-sdk-did-auth-siop-authenticator'
+
+const agent = createAgent<IDidAuthSiopOpAuthenticator & IResolver>({
+  plugins: [
+    new DidAuthSiopOpAuthenticator(),
+    new DIDResolverPlugin({
+      resolver: new Resolver({
+        ...ethrDidResolver({ infuraProjectId: INFURA_PROJECT_ID }),
+        ...webDidResolver(),
```

Comment:
I would like to include our UniResilver in the examples. Also a didkey makes more sense then the ethrDidResolver propbably, unless we can ommit the INFURA stuff. We shouldn't be needing that anyway, since basic DIDs using the ethr driver do not need actual on chain state

Reply:
changed